course_discovery/apps/edly/api/v1/views.py

```diff
@@ -0,0 +1,104 @@
+"""
+Views for API v1.
+"""
+from django.utils.text import slugify
+from rest_framework import (
+    status,
+    viewsets
+)
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from course_discovery.apps.edly.api.v1.serializers import EdlyProgramSerializer
+from course_discovery.apps.course_metadata.models import (
+    Course,
+    CourseRun,
+    Organization,
+    Program
+)
+
+
+class ProgramViewSet(viewsets.ModelViewSet):
+    """
+    Programs Create & Update.
+    """
+    serializer_class = EdlyProgramSerializer
+    permission_classes = (IsAuthenticated,)
+
+    def prepare_and_set_read_only_data(self, data, context):
+        """
+        Extracts read only data from data dictionary and sets it in context.
+        """
+        if data.get('org'):
+            organization = Organization.objects.filter(key=data['org']).first()
+            context['authoring_organization'] = organization
+            context['credit_backing_organization'] = organization
+
+        if data.get('course_runs'):
+            courses = []
+            excluded_course_runs = []
+            for course_run_key in data['course_runs']:
+                course_run = CourseRun.objects.filter(key=course_run_key).first()
+                course = course_run.course
+                courses.append(course)
+
+                for run in course.course_runs.exclude(key=course_run_key):
+                    excluded_course_runs.append(run)
+
+            # Dedupe courses & excluded course runs
+            courses = list(set(courses))
+            excluded_course_runs = list(set(excluded_course_runs))
```

Comment:
If you are trying to remove duplicates, u can add a `.distinct()` in the above approach.